### PR TITLE
support dumping binary data in hexadecimal representation

### PR DESCRIPTION
**Pull request by tangenta**

### CODE REVIEW

File: v4/export/ir.go [hunk 1]
```diff
@@ -19,6 +21,15 @@ type SQLRowIter interface {
...
 }
 
+type RowReceiverStringer interface {
+	RowReceiver
+	Stringer
+}
+
+type Stringer interface {
+	ToString() string
+}
```
Comment (kennytm) [inline v4/export/ir.go]: why not reuse `fmt.Stringer`?

File: v4/export/test_util.go [hunk 2]
```diff
@@ -71,65 +74,67 @@ func newMockMetaIR(targetName string, meta string, specialComments []string) Met
...
-		idx:  0,
-		data: m.data,
+func newMockTableIR(databaseName, tableName string, data [][]interface{}, specialComments, colTypes []string) TableDataIR {
+	values := make([][]driver.Value, len(data))
+	for i := range values {
+		values[i] = make([]driver.Value, len(data[i]))
+		for j := range data[i] {
+			values[i][j] = data[i][j]
+		}
+	}
```
Comment (kennytm) [inline v4/export/test_util.go]: could `values := data` work? 🤔

File: v4/export/writer_util.go [hunk 3]
```diff
@@ -108,37 +86,127 @@ func wrapStringWith(str string, wrapper string) string {
...
+	"DATE":      SQLTypeStringMaker,
+	"TIME":      SQLTypeStringMaker,
+	"YEAR":      SQLTypeStringMaker,
+
+	"BINARY":    SQLTypeBytesMaker,
+	"VARBINARY": SQLTypeBytesMaker,
+	"BLOB":      SQLTypeBytesMaker,
+	"BIT":       SQLTypeBytesMaker,
+
+	"INT":     SQLTypeIntegerMaker,
```
Comment (kennytm) [inline v4/export/writer_util.go]: Can this handle MEDIUMBLOB / BIGINT / etc?

File: v4/export/writer_util.go [hunk 4]
```diff
@@ -108,37 +86,127 @@ func wrapStringWith(str string, wrapper string) string {
...
+}
+func (s *SQLTypeString) ReportSize() uint64 {
+	if s.Valid {
+		return uint64(len(s.String))
+	}
+	return uint64(len("NULL"))
+}
+func (s *SQLTypeString) ToString() string {
+	if s.Valid {
+		return fmt.Sprintf(`'%s'`, s.String)
```
Comment (kennytm) [inline v4/export/writer_util.go]: What if the string contains a `'` (e.g. `'this shouldn''t fail'`)

File: v4/export/writer_util.go [hunk 5]
```diff
@@ -108,37 +86,127 @@ func wrapStringWith(str string, wrapper string) string {
...
+	"TIME":      SQLTypeStringMaker,
+	"YEAR":      SQLTypeStringMaker,
+
+	"BINARY":    SQLTypeBytesMaker,
+	"VARBINARY": SQLTypeBytesMaker,
+	"BLOB":      SQLTypeBytesMaker,
+	"BIT":       SQLTypeBytesMaker,
+
+	"INT":     SQLTypeIntegerMaker,
+	"FLOAT":   SQLTypeIntegerMaker,
```
Comment (kennytm) [inline v4/export/writer_util.go]: FLOAT is not an "integer" 🙃

File: v4/export/ir.go (hunk 1 above)
Comment (tangenta) [reply v4/export/ir.go]: There is a naming collide between `fmt.Stringer`'s `String()` and `sql.NullString`'s `String` field.

File: v4/export/test_util.go (hunk 2 above)
Comment (tangenta) [reply v4/export/test_util.go]:
```
cannot use values (type [][]interface {}) as type [][]driver.Value in field value
```

File: v4/export/writer_util.go (hunk 3 above)
Comment (tangenta) [reply v4/export/writer_util.go]: Sure, we can add fill this map gradually.

File: v4/export/test_util.go (hunk 2 above)
Comment (kennytm) [reply v4/export/test_util.go]: Could `data` be `[][]driver.Value` instead of `[][]interface{}`?

File: v4/export/writer_util.go [hunk 6]
```diff
@@ -191,12 +191,16 @@ func (s *SQLTypeString) ReportSize() uint64 {
...
 	if s.Valid {
-		return fmt.Sprintf(`'%s'`, s.String)
+		return fmt.Sprintf(`'%s'`, escapeSingleQuote(s.String))
 	} else {
 		return "NULL"
 	}
 }
 
+func escapeSingleQuote(src string) string {
+	return strings.ReplaceAll(src, "'", "\\'")
```
Comment (kennytm) [inline v4/export/writer_util.go]: The SQL-compliant escaping is `''` not `\'`. 

And if we need to support `\'` we gotta escape `\` as `\\` too.

File: v4/export/writer_util.go (hunk 3 above)
Comment (kennytm) [reply v4/export/writer_util.go]: Ok.

**Pull request merged**